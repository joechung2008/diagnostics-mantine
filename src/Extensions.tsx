import { Button, Text } from "@mantine/core";
import { byKey, isExtensionInfo, toNavLink } from "./utils";

const Extensions: React.FC<ExtensionsProps> = ({ extensions, onLinkClick }) => {
  const links = Object.values(extensions)
    .filter(isExtensionInfo)
    .map(toNavLink)
    .sort(byKey);

  return (
    <nav className="extension-root" aria-label="Extensions">
      {links.map((link) => (
        <Button
          key={link.key}
          className="extension-nav-button"
          variant="subtle"
          onClick={(e) => onLinkClick?.(e, link)}
        >
          <Text size="sm">{link.name}</Text>
        </Button>
      ))}
    </nav>
  );
};

export default Extensions;
