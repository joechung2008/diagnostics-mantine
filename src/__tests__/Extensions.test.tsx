import { fireEvent, screen } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";
import Extensions from "../Extensions";
import { render } from "./test-utils";

describe("Extensions", () => {
  it("should render buttons for ExtensionInfo", () => {
    const extensions = {
      ext1: { extensionName: "ext1" } as ExtensionInfo,
      ext2: {
        lastError: { errorMessage: "error", time: "time" },
      } as ExtensionError,
    };
    const onLinkClick = vi.fn();
    const { container } = render(
      <Extensions extensions={extensions} onLinkClick={onLinkClick} />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should call onLinkClick when button is clicked", () => {
    const extensions = {
      ext1: { extensionName: "ext1" } as ExtensionInfo,
    };
    const onLinkClick = vi.fn();
    render(<Extensions extensions={extensions} onLinkClick={onLinkClick} />);

    fireEvent.click(screen.getByText("ext1"));
    expect(onLinkClick).toHaveBeenCalled();
  });
});
