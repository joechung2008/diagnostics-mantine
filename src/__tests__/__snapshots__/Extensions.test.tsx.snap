// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Extensions > should render buttons for ExtensionInfo 1`] = `
<style
  data-mantine-styles="true"
>
  




</style>
`;
